build.rs

```diff
@@ -1,30 +1,45 @@
 #![recursion_limit = "1024"]
 #[macro_use]
 extern crate error_chain;
+#[macro_use]
+extern crate quote;
+extern crate syn;
 extern crate walkdir;
 
-use walkdir::WalkDir;
+use std::env;
 use std::fmt;
 use std::fs::File;
+use std::fs;
 use std::io::{Write, BufWriter, stderr};
+use std::path::Path;
 use std::process::exit;
 
+use quote::Ident;
+use walkdir::WalkDir;
+
 error_chain! {
     foreign_links {
         Io(::std::io::Error);
         WalkDir(::walkdir::Error);
     }
 }
 
+const SOURCE_TEST_DIR: &str = "tests/source";
+
 // Location of the dist folder
 const DIST: &str = "frontend/dist/";
 
 // Where to write the Assets out to
 const ASSETOUT: &str = "src/asset.in";
 
-/// Find the assets and write out the corresponding files
-pub fn main() {
-    if let Err(ref e) = write_asset_file() {
+fn run() -> Result<()> {
+    generate_source_tests()?;
```

Comment:
We could probably run these tasks in parallel join the threads then return any errors that occur.